### DISCUSSION TITLE
why webrtc-streamer use STUN/TURN ?

### DISCUSSION THREAD

**Discussion by randall05** [mpromonet/webrtc-streamer#695] [Q&A]
i want view the camera in browser in real time by RTSP, so the browser must can connect to Webrtc-streamer, and webrtc-streamer must can connect to the Camera.

in this case, i think the webrtc-streamer is like a proxy, so can I disable STUN feature by "-s-" ?

**Comment by mpromonet** [comment]
Hi,

If there is no NAT between your browser and webrtc-streamer STUN/TURN is not needed.

Best Regards 
Michel

**Comment by randall05** [comment]
in your sample code "WebRtcStreamer.html":

       `this.webRtcServer = new WebRtcStreamer("video", webrtcConfig.url);`

i think the Parameter “webrtcConfig.url” is the address of webrtc-streamer.

My question is :
if the browser can't connect webrtc-streamer directly which behind a router/firewall, i need a STUN/TURN services now.
but what is "webrtcConfig.url" now ? the webrtc-streamer address or the STUN/TURN service address?

**Reply by mpromonet** [reply]
Hi,

This is the URL of webrtc-streamer api, this allow to serve HTML/js from another http server. This is not stun/TURN URL.

Best Regards 
Michel

**Reply by randall05** [reply]
thank you very much 💯

**Comment by randall05** [comment]
we define the STUN/TURN service url in webrtc-streamer. if browser can't connect webrtc-streamer directly, it can't get the STUN/TURN service url by the webrtc-streamer api, right?

if browser can get the STUN/TURN service url which defined in webrtc-streamer (by Router/firewall settings likely), that's mean the browser can connect webrtc-streamer already, so why it need a STUN/TURN service now?

i know the browser can use a specified thrid-party STUN/TURN service url, but where can i define the url in HTML/JS code in order to  let brower know it need make a P2P connection with a specified webrtc-streamer which behind a router/firewall and use same STUN/TURN service?

**Reply by mpromonet** [reply]
Hi @randall05 

The STUN/TURN configuration is done on server side using argument -s/-S/-t/-T and browser side  code get it using /api/getIceServers. Then both side share the same STUN/TURN configuration.

Best Regards,
Michel.

**Reply by randall05** [reply]
thanks for your replying ~:)

so browser side must get STUN/TURN configuration by "/api/getIceServers". 

if browser can already connect webrtc-streamer, why do we need STUN/TURN ?

**Reply by mpromonet** [reply]
Hi @randall05 

This is not the same interface, /api is http interface mainly for webrtc signaling, stun/TURN is for choosing ICE candidate.

Best Regards 
Michel
